### PR TITLE
Issues #11, #12, #13, and #14

### PR DESCRIPTION
**Pull request by jayhilden**
This pull request covers issues #11 #12, #13, #14

### CODE REVIEW

File: Elmah.Io.ElasticSearch/ExtensionMethods.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        public static string GetRequestString<T>(this ISearchResponse<T> response) where T : class
+        {
+            var request = response.RequestInformation.Request;
+            return Encoding.Default.GetString(request);
+        }
+
+        /// <summary>
+        /// ensure that the ES command was run successfully on the server
+        /// </summary>
+        public static void VerifySuccessfulResponse(this IResponse response)
```
Comment (jayhilden) [inline Elmah.Io.ElasticSearch/ExtensionMethods.cs]: I added this helper method VerifySuccessfulResponse() that can be used against any ES call to ensure it was successful.

File: README.md [hunk 2]
```diff
@@ -16,13 +16,26 @@ Replace the connection string URL with your ElasticSearch URL and add the elmah
 
 That's it, dudes!
 
-If you don't like the default generated index named _elmah_, you should override it by specifying a default index name:
-
-    <errorLog
-        type="Elmah.Io.ElasticSearch.ElasticSearchErrorLog, Elmah.Io.ElasticSearch"
-        connectionStringName="ElmahIoElasticSearch"
-        defaultIndex="myindex" />
+The Elastic index name can be specified one of two ways, let's say you want the index name to be elmahCurrent:
```
Comment (jayhilden) [inline README.md]: updated documentation, we implemented this easier way to specify the ES index name a while ago but never updated the documentation.

**Comment by jayhilden** [comment]
@ThomasArdal Elmah.ElasticSearch is no longer available on Nuget.  Was it removed for some reason?

**Comment by ThomasArdal** [comment]
Sorry I accidentally removed the nuget package. It's there again.

**Comment by ThomasArdal** [comment]
@jayhilden You have been added as a collaborator on this project. You should be able to merge the pull request.

**Comment by jayhilden** [comment]
Thanks for the promotion @ThomasArdal!  I'm not able to merge the pull request yet though:

image

**Comment by ThomasArdal** [comment]
GitHub tells me, that you haven't accepted the invitation yet. You should
have write access after you accept.

**Comment by jayhilden** [comment]
Thanks @ThomasArdal, I was able to get in and merge the branch into master.

Will this automatically build in TeamCity somewhere?  Also, will it automatically rev and push out to nuget?  If you can show me how to tackle those two things I think I'll be good to go with everything.

**Comment by ThomasArdal** [comment]
Yeah. It's build on teamcity.codebetter.com and automatically pushed to nuget.org, if everything builds and unit tests are green.

You can find the project on TeamCity here: http://teamcity.codebetter.com/viewType.html?buildTypeId=bt1123

**Comment by jayhilden** [comment]
@ThomasArdal That's really slick, I'll update the README.md file with the build status and nuget information when I get time.

I think the only other thing then is how do the dependencies get updated in Nuget?

image

**Comment by ThomasArdal** [comment]
There's a nuspec file for that:

https://github.com/elmahio/Elmah.Io.ElasticSearch/blob/master/Elmah.Io.ElasticSearch/Elmah.Io.ElasticSearch.nuspec

**Comment by jayhilden** [comment]
Thank you, I'll update that soon.  Are you good with me creating a permanent development branch as well so that not every commit gets pushed to NUGET?  I'd also like to update team city and create a 1.1 release when all this is complete.

**Comment by ThomasArdal** [comment]
I think that every commit on master should build a new NuGet package. If
you wan't to work against a 1.1 release, why not just create a branch for
that and merge it to master when ready? You can even setup automatic push
of a 1.1 prerelease nuget package from the 1.1 branch.

**Comment by jayhilden** [comment]
Yep, you and I are on exactly the same page.  Where would I setup the prerelease push to nuget?

**Comment by ThomasArdal** [comment]
I can set it up on teamcity.codebetter.com, if you send the branch name.
Also remember to update the nuspec file on that branch to include the
prerelease tag.

**Pull request merged**